packages/marshal/src/make-far.js

```diff
@@ -199,18 +201,121 @@ export const Remotable = (
 };
 harden(Remotable);
 
+/**
+ * Wrap function with defensive layer for hardening and checking the
+ * args and outcome (result or error).
+ *
+ * @param {(...args: any[]) => any} func
+ * @param {FarOptions=} options
+ * @returns {(...args: any[]) => any}
+ */
+const wrapFuncDefensively = (func, { allowNonPassables = false } = {}) => {
+  const name = func.name;
+  const wrapper = (...args) => {
+    let result;
+    for (const arg of args) {
+      harden(arg);
+      if (!allowNonPassables) {
```

Comment:
Rather than checking `allowNonPassables` on every call and return, how about just returning the original, unwrapped function if `allowNonPassables` is `true`?